src/hf_hydrodata/point.py

```diff
@@ -1094,6 +1119,9 @@ def _get_point_citations(dataset):
             "Source: https://ameriflux.lbl.gov/data/data-policy/"
         )
 
+    elif dataset == "jasechko_2024":
```

Comment:
Not super relevant to this PR, but I think this whole block could be simplified doing something like
```python
match dataset:
  "usgs_nwis":
    return "citation stuff..."
 "jascechko_2024":
    return "citation stuff..."
  ...
```